oak_attestation_verification/src/amd.rs

```diff
@@ -0,0 +1,199 @@
+use sha2::Sha384;
+use x509_cert::{
+    der::{referenced::OwnedToRef, Encode},
+    Certificate, Version,
+};
+use zerocopy::{AsBytes, FromZeroes};
+
+// The keys in the key-value map of X509 certificates are Object Identifiers
+// (OIDs) which have a global registry. The present OIDs are taken from
+// Table 8 of
+// https://www.amd.com/content/dam/amd/en/documents/epyc-technical-docs/specifications/57230.pdf
+const RSA_SSA_PSS_OID: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.2.840.113549.1.1.10");
+const PRODUCT_NAME_OID: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.3.6.1.4.1.3704.1.2");
+const BL_OID: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.3.6.1.4.1.3704.1.3.1");
+const TEE_OID: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.3.6.1.4.1.3704.1.3.2");
+const SNP_OID: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.3.6.1.4.1.3704.1.3.3");
+const CHIP_ID_OID: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.3.6.1.4.1.3704.1.4");
+const UCODE_OID: ObjectIdentifier = ObjectIdentifier::new_unwrap("1.3.6.1.4.1.3704.1.3.8");
+
+// Verifies validity of a matching ARK, ASK certificate pair.
+//
+// Validate at least a subset of Appendix B.3 of
+// https://www.amd.com/content/dam/amd/en/documents/epyc-technical-docs/programmer-references/55766_SEV-KM_API_Specification.pdf
+// Ideally, we'd check everything listed there.
+pub fn validate_ark_ask_certs(ark: &Certificate, ask: &Certificate) -> anyhow::Result<()> {
+    anyhow::ensure!(
+        ark.tbs_certificate.version == Version::V3,
+        "unexpected version of ARK cert"
+    );
+    anyhow::ensure!(
+        ask.tbs_certificate.version == Version::V3,
+        "unexpected version of ASK cert"
+    );
+
+    verify_cert_signature(ark, ask)?;
+    verify_cert_signature(ark, ark)?;
+
+    Ok(())
+}
+
+pub fn verify_cert_signature(signer: &Certificate, signee: &Certificate) -> anyhow::Result<()> {
+    anyhow::ensure!(
+        signee.signature_algorithm.oid == RSA_SSA_PSS_OID,
+        "unsupported signature algorithm: {:?}",
+        signee.signature_algorithm
+    );
+
+    let verifying_key = {
+        let pubkey_info = signer
+            .tbs_certificate
+            .subject_public_key_info
+            .owned_to_ref();
+        let pubkey = RsaPublicKey::try_from(pubkey_info)
+            .map_err(|_err| anyhow::anyhow!("could not parse RSA public key"))?;
+        rsa::pss::VerifyingKey::<Sha384>::new(pubkey)
+    };
+
+    let message = signee
+        .tbs_certificate
+        .to_der()
+        .map_err(|_err| anyhow::anyhow!("could not extract message to verify RSA signature"))?;
+    let signature = Signature::try_from(signee.signature.raw_bytes())
+        .map_err(|_err| anyhow::anyhow!("could not extract RSA signature"))?;
+
+    Ok(verifying_key
+        .verify(&message, &signature)
+        .map_err(|_err| anyhow::anyhow!("signature verification failed"))?)
+}
+
+// Currently unused, use `pub` only to disable the warning.
```

Comment:
Which warning? If that's the unused warning, then it should be suppressed in some other way, not by exposing a function that should not be exposed.